From 745b48f3dd8eeaa0a99c1911dcc565f51d2a9122 Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ilies.chergui@gmail.com>
Date: Sat, 8 Apr 2023 15:49:28 +0100
Subject: [PATCH 3/5] lib/tools-common.in: keystoretool -> nvluks-srv-app

Swap keystoretool for nvluks-srv-app, with DISK_ENCRYPTION_CONTEXT
placeholder

Signed-off-by: Ilies CHERGUI <ilies.chergui@gmail.com>
---
 lib/tools-common.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/tools-common.in b/lib/tools-common.in
index 65c0fd1..749b1fa 100644
--- a/lib/tools-common.in
+++ b/lib/tools-common.in
@@ -738,7 +738,7 @@ initialize_devices() {
     local fakept=`mktemp -q`
     if make_fake_parttable $fakept; then
 	local ppdir=`mktemp -q -d /run/setup-pp.XXXXXX`
-	keystoretool -p > $ppdir/passphrase
+	/usr/sbin/nvluks-srv-app --get-unique-pass --context-string=@DISK_ENCRYPTION_CONTEXT@ > $ppdir/passphrase
 	if ! create_luks_partitions $ppdir $fakept; then
 	    rm -rf $ppdir
 	    return 1
-- 
2.25.1

